rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/edge/TbMsgPushToEdgeNode.java

```diff
@@ -84,173 +69,92 @@
+    String getIgnoredMessageSource() {
+        return DataConstants.EDGE_MSG_SOURCE;
     }
 
-    private void processMsg(TbContext ctx, TbMsg msg) {
+    @Override
+    protected Class<TbMsgPushToEdgeNodeConfiguration> getConfigClazz() {
+        return TbMsgPushToEdgeNodeConfiguration.class;
+    }
+
+    @Override
+    protected void processMsg(TbContext ctx, TbMsg msg) {
         if (EntityType.EDGE.equals(msg.getOriginator().getEntityType())) {
-            EdgeEvent edgeEvent = buildEdgeEvent(msg, ctx);
+            EdgeEvent edgeEvent = buildEvent(msg, ctx);
             if (edgeEvent != null) {
                 EdgeId edgeId = new EdgeId(msg.getOriginator().getId());
                 notifyEdge(ctx, msg, edgeEvent, edgeId);
+            } else {
+                tellFailure(ctx, msg);
             }
         } else {
             PageLink pageLink = new PageLink(DEFAULT_PAGE_SIZE);
             PageData<EdgeId> pageData;
+            boolean edgeNotified = false;
             do {
                 pageData = ctx.getEdgeService().findRelatedEdgeIdsByEntityId(ctx.getTenantId(), msg.getOriginator(), pageLink);
                 if (pageData != null && pageData.getData() != null && !pageData.getData().isEmpty()) {
                     for (EdgeId edgeId : pageData.getData()) {
-                        EdgeEvent edgeEvent = buildEdgeEvent(msg, ctx);
-                        if (edgeEvent == null) {
-                            log.debug("Edge event type is null. Entity Type {}", msg.getOriginator().getEntityType());
-                            ctx.tellFailure(msg, new RuntimeException("Edge event type is null. Entity Type '" + msg.getOriginator().getEntityType() + "'"));
-                        } else {
+                        EdgeEvent edgeEvent = buildEvent(msg, ctx);
+                        if (edgeEvent != null) {
                             notifyEdge(ctx, msg, edgeEvent, edgeId);
+                            edgeNotified = true;
+                        } else {
+                            tellFailure(ctx, msg);
                         }
                     }
                     if (pageData.hasNext()) {
                         pageLink = pageLink.nextPageLink();
                     }
                 }
             } while (pageData != null && pageData.hasNext());
+
+            if (!edgeNotified) {
+                // ack in case no edges are related to provided entity
+                ctx.ack(msg);
+            }
         }
     }
 
+    private void tellFailure(TbContext ctx, TbMsg msg) {
+        String errMsg = String.format("Edge event type is null. Entity Type %s", msg.getOriginator().getEntityType());
+        log.warn(errMsg);
+        ctx.tellFailure(msg, new RuntimeException(errMsg));
+    }
+
     private void notifyEdge(TbContext ctx, TbMsg msg, EdgeEvent edgeEvent, EdgeId edgeId) {
         edgeEvent.setEdgeId(edgeId);
         ctx.getEdgeEventService().save(edgeEvent);
         ctx.tellNext(msg, SUCCESS);
```

Comment:
One incoming message may produce multiple outcoming messages. Basically, you send the same message for each Edge using "tellNext". This is wrong since Rule Engine will be confused and will not track the delivery of each message. You should use "TbContext.enqueue" instead and maybe change the originator of the message using 
```
TbMsg.newMsg(String type, EntityId originator, TbMsgMetaData metaData, String data, TbMsgCallback callback)
```